File: annotationdemo/src/main/java/com/example/annotationdemo/MainActivity.java
package com.example.annotationdemo;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import com.example.annotaionprocessor.BindView;

public class MainActivity extends AppCompatActivity {

    @BindView(R.id.text)
    TextView title;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Log.d(TAG, "onCreate: "+title);

    }

    private static final String TAG = "MainActivity";
}

File: myapplication2/src/main/java/com/example/myapplication2/MainActivity.kt
package com.example.myapplication2

import android.os.Bundle
import android.util.Log
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import java.io.File
import kotlin.random.Random

private const val TAG = "MainActivity"

class MainActivity : AppCompatActivity() {
    var directoryObserver: DirectoryObserver? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        directoryObserver = DirectoryObserver(cacheDir.absolutePath)
        directoryObserver!!.start(object : DirectoryObserver.IEventChangedListener {
            override fun onChange(path: String, event: DirectoryObserver.FileEvent) {
                Log.d(TAG, "onChange() called with: path = $path, event = $event")
            }
        })
        findViewById<Button>(R.id.btn_send).setOnClickListener {
            val file = File(cacheDir.absolutePath+ File.separator+"1/1.md")
            if (file.exists()){
                val text = Random(4343).toString()
                file.appendText(text,Charsets.UTF_8)
            }
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        directoryObserver?.stop()
    }
}